### PR TITLE
Refactor BaseFold test code

### PR DESCRIPTION
**Pull request by yczhangsjtu**
Extracting small PR from #294 

Refactor the test code in `mpcs/src/lib.rs`. Define some utility functions to simplify the tests.

Also merged the test functions in `mpcs/src/basefold.rs`.

### CODE REVIEW

File: mpcs/src/basefold.rs [hunk 1]
```diff
@@ -1191,108 +1191,53 @@ mod test {
...
-        // Both challenge and poly are over base field
-        run_batch_commit_open_verify::<GoldilocksExt2, PcsGoldilocksBaseCode>(true, 10, 11);
-    }
-
-    #[test]
-    fn batch_commit_open_verify_goldilocks_rscode_base() {
-        // Both challenge and poly are over base field
-        run_batch_commit_open_verify::<GoldilocksExt2, PcsGoldilocksRSCode>(true, 10, 11);
+    fn commit_open_verify_goldilocks() {
+        for base in [true, false].into_iter() {
```
Comment (matthiasgoergens) [inline mpcs/src/basefold.rs:1195]: Small cosmetic suggestion:

```suggestion
        for base in [true, false] {
```

(Similar for the other loops, too.)

File: mpcs/src/lib.rs [hunk 2]
```diff
@@ -359,15 +359,79 @@ fn err_too_many_variates(function: &str, upto: usize, got: usize) -> Error {
...
+        num_vars: usize,
+    ) -> (Pcs::ProverParam, Pcs::VerifierParam) {
+        let poly_size = 1 << num_vars;
+        let param = Pcs::setup(poly_size).unwrap();
+        Pcs::trim(param, poly_size).unwrap()
+    }
+
+    pub fn gen_rand_poly<E: ExtensionField>(
+        num_vars: usize,
+        base: bool,
```
Comment (matthiasgoergens) [inline mpcs/src/lib.rs:382]: Didn't we have a PR where we moved away from using booleans here (and in similar situations)?

**Review by matthiasgoergens — APPROVED** [review]
Looks ok in general, but please have a look at my comments.  Thanks!

File: mpcs/src/lib.rs (hunk 2 above)
Comment (matthiasgoergens) [reply mpcs/src/lib.rs:382]: Hmm, I guess that's happening in the follow up PR?

Comment (lispc) [reply mpcs/src/lib.rs:382]: @yczhangsjtu feel free either to resolve coversation (so i can merge) or change some code.

**Pull request merged**